File: server/app/schemas/system.py
from typing import List
from pydantic import BaseModel

class UploadSettings(BaseModel):
    max_size_mb: int
    allowed_extensions: List[str]

class AccessSettings(BaseModel):
    allow_external_ip: bool = False

class StorageSettings(BaseModel):
    data_path: str = "data"
    current_abs_path: str = ""
    migration_action: str = "none" # none, migrate, create_new

class GeneralSettings(BaseModel):
    project_name: str = "EasyStore"
    port: int = 8000

class SystemSettings(BaseModel):
    upload: UploadSettings
    access: AccessSettings = AccessSettings()
    storage: StorageSettings = StorageSettings()
    general: GeneralSettings = GeneralSettings()

File: web/src/store/system.ts
import { defineStore } from 'pinia'
import request from '@/api/request'

interface SystemSettings {
  upload: {
    max_size_mb: number
    allowed_extensions: string[]
  }
  access: {
    allow_external_ip: boolean
  }
  storage: {
    data_path: string
  }
  general: {
    project_name: string
    port: number
  }
}

export const useSystemStore = defineStore('system', {
  state: () => ({
    settings: {
      upload: {
        max_size_mb: 5,
        allowed_extensions: ['jpg', 'jpeg', 'png', 'gif', 'webp']
      },
      access: {
        allow_external_ip: false
      },
      storage: {
        data_path: 'data'
      },
      general: {
        project_name: 'EasyStore',
        port: 8000
      }
    } as SystemSettings,
    isLoaded: false
  }),
  actions: {
    async fetchSettings() {
      try {
        const res: any = await request.get('/system/settings')
        this.settings = res
        this.isLoaded = true
      } catch (error) {
        console.error('Failed to fetch system settings', error)
      }
    }
  }
})
